docs/streaming-programming-guide.md

```diff
@@ -644,13 +644,44 @@ methods for creating DStreams from files as input sources.
     </div>
     </div>
 
-	Spark Streaming will monitor the directory `dataDirectory` and process any files created in that directory (files written in nested directories not supported). Note that
-
-     + The files must have the same data format.
-     + The files must be created in the `dataDirectory` by atomically *moving* or *renaming* them into
-     the data directory.
-     + Once moved, the files must not be changed. So if the files are being continuously appended, the new data will not be read.
-
+	Spark Streaming will monitor the directory `dataDirectory` and process any files created in that directory.
+
+     + A simple directory can be supplied, such as `hdfs://namenode:8040/logs/`.
+       All files directly such a path will be processed as they are discovered.
+     + A regular expression can be supplied instead, such as
```

Comment:
Also if I read the code correctly, it always treats the input as a glob. If so I think that's how it should be documented. That is, if you give it a path to a file containing "?" it's not going to do what you think. The glob isn't optional if I read it right.

But yeah this looks good to me code-wise.